@garymathews , On Android 8.0 emulator I am getting this runtime error. Strangely, with master I don't see any issues.

```
[ERROR] :  TiExceptionHandler: (main) [13229,13229] ----- Titanium Javascript Runtime Error -----
[ERROR] :  TiExceptionHandler: (main) [1,13230] - In /app.js:49,14
[ERROR] :  TiExceptionHandler: (main) [0,13230] - Message: Uncaught Attempt to invoke virtual method 'android.content.res.Resources android.content.Context.getResources()' on a null object reference
[ERROR] :  TiExceptionHandler: (main) [0,13230] - Source:   drawerView.setRightView(realRight);
[ERROR] :  V8Exception: Exception occurred at /app.js:49: Uncaught Attempt to invoke virtual method 'android.content.res.Resources android.content.Context.getResources()' on a null object reference
[ERROR] :  V8Exception: Attempt to invoke virtual method 'android.content.res.Resources android.content.Context.getResources()' on a null object reference
```

--- a/android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java
+++ b/android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java
@@ -502,10 +502,10 @@ public abstract class TiViewProxy extends KrollProxy implements Handler.Callback
 			}
 
 			Activity activity = getActivity();
-			if (activity.isDestroyed()) {
+			if (activity != null && activity.isDestroyed()) {
 				activity = activity.getParent();
 			}
-			if (activity == null) {
+			if (activity == null || activity.isDestroyed()) {
 				activity = TiApplication.getAppRootOrCurrentActivity();
 			}
 			view = createView(activity);

--- a/android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java
+++ b/android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java
@@ -483,8 +483,7 @@ public abstract class TiViewProxy extends KrollProxy implements Handler.Callback
 	 */
 	public TiUIView getOrCreateView()
 	{
-		TiBaseActivity activity = (TiBaseActivity) getActivity();
-		if (activity == null || activity.isDestroyed() || view != null) {
+		if (view != null) {
 			return view;
 		}
 
@@ -503,6 +502,12 @@ public abstract class TiViewProxy extends KrollProxy implements Handler.Callback
 			}
 
 			Activity activity = getActivity();
+			if (activity.isDestroyed()) {
+				activity = activity.getParent();
+			}
+			if (activity == null) {
+				activity = TiApplication.getAppRootOrCurrentActivity();
+			}
 			view = createView(activity);
 			if (isDecorView) {
 				if (activity != null) {

